src/modules/processing/matchActionDecider.js

```diff
@@ -76,8 +77,13 @@ export default function(r, host, userFormat, isAudioOnly, lang, isAudioMuted, di
                     params = {
                         type: pickerType,
                         picker: r.picker,
-                        u: Array.isArray(r.urls) ? r.urls[1] : r.urls,
-                        copy: audioFormat === "best" ? true : false
+                        u: createStream({
```

Comment:
`audioFormat` isn't passed. downloading audio from picker in best & mp3 format results in a file with no file extension:
<img width="316" alt="image" src="https://github.com/imputnet/cobalt/assets/71202418/0b508225-25e2-48f6-a7cb-9dc679e51391">

downloading in any format other than best leads to a crash, because you're using `pickerType` as **stream** type. they're not equal. cobalt ends up defaulting to video "render" stream type, because `isAudioOnly` isn't set either.

were you trying to make an internal stream that'd use cookies here?

note: audio format must be set to user preference, but you should match best available audio to it. if user prefers mp3, and best available (`r.bestAudio`) is mp3, then stream type should be "bridge". if it needs converting, then it should be "render".